src/nvim/eval.c

```diff
@@ -16682,16 +16631,16 @@ static void f_termopen(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   (void)setfname(curbuf, (char_u *)buf, NULL, true);
   // Save the job id and pid in b:terminal_job_{id,pid}
   Error err = ERROR_INIT;
+  dict_set_var(curbuf->b_vars, cstr_as_string("terminal_channel_id"),
+               INTEGER_OBJ(chan->id), false, false, &err);
```

Comment:
Missing call to `api_clear_error(&err);`. If an error is set here and in the next `dict_set_var()`, we have a memory leak.